akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -726,3 +729,139 @@ private[akka] object Log {
   private final val OffInt = LogLevels.Off.asInt
   private final val DefaultLogLevels = LogLevels(onElement = Logging.DebugLevel, onFinish = Logging.DebugLevel, onFailure = Logging.ErrorLevel)
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] object TimerKeys {
+  case object TakeWithinTimerKey
+  case object DropWithinTimerKey
+  case object GroupedWithinTimerKey
+}
+
+/**
+ * INTERNAL API
+ */
+private[stream] final case class GroupedWithin[T](n: Int, d: FiniteDuration) extends TimerStage[T, immutable.Seq[T]] {
+  import TimerKeys._
+  private var buf: Vector[T] = Vector.empty
+
+  override def preStart(ctx: AsyncContext[immutable.Seq[T], Scheduled]) = {
+    super.preStart(ctx)
+    schedulePeriodically(GroupedWithinTimerKey, d)
```

Comment:
This is why I like the GraphStage better, since you are not forced to use only one callback (onAsync). The problem if you add the thing to AsyncStage itself, then if people forget to call `super.onAsync` then the timer event will be lost.

`preStart` and `postStop` are already a problem of course.